ControllerConnection/ControllerConnection.py

```diff
@@ -10,30 +9,31 @@ class ControllerConnection:
 
     #   The type of update,  The Mixer, App index   Associated value from the application
     _actions = {
-        ChangedValue.VOLUME: lambda subject, index: subject.applications[index].volume,
+        ChangedValue.VOLUME: lambda subject, index: int(subject.applications[index].volume),
         ChangedValue.COLOR: lambda subject, index: subject.applications[index].color_matrix,
     }
 
     def __init__(self, volume_mixer):
-        self.volume_mixer = volume_mixer
+        self._volume_mixer = volume_mixer
         # TODO: Figure out how to set the port dynamically.
         self.serial = serial.Serial('COM4', 9600)
         ControllerReceiver(self).start()
 
     def send_to_controller(self, control, action, value):
-        payload = str(control) + "." + str(action) + "." + str(value)
+        payload = str(control) + "." + str(action.value) + "." + str(value)
         self.serial.write(payload.encode())
 
     def receive_from_controller(self):
         controller_msg = self.serial.readline()
-        data = controller_msg.decode("utf-8").split(".")
-        self.volume_mixer.modify_application(data[0], data[1], data[2])
+        data = controller_msg.decode("utf-8").strip().split(".")
+        self._volume_mixer.modify_application(int(data[0]), ChangedValue(int(data[1])), int(data[2]))
 
     def connection_alive(self):
         com_ports = [tuple(p) for p in list(serial.tools.list_ports.comports())]
         for com_port in com_ports:
             if self.BOARDNAME in com_port:
                 return True
+
```

Comment:
Not sure if it's a mobile formatting thing or what but why do we have return true then false?